### PR TITLE
Run test as monorepo OR normal, not both

### PR DESCRIPTION
**Pull request by scruffystuffs**
# Overview

Previously, we tried running both normal build polling and monorepo polling at the same time.  However this causes us to report incorrect errors during normal build failures.  This PR attempts to detect the correct project type, and fall back to normal builds unless the user tells us to fall back to normal builds.

Requires fossas/FOSSA#6311, which has been deployed in cloud in version `3.34.19`

## Acceptance criteria

* `fossa test` and `fossa report` use the same logic to detect project types.
* We don't hide or obscure errors in build polling.

## Testing plan

* Run analyze/test/report for a normal project 
* Run `test` and `report` on a monorepo project, against prod.
  * This will not work on prod until the referenced PR is deployed.
* Run `test` and `report` on a monorepo project, against a core version without the `/api/cli/:locator/project` route.
  * This will work on prod UNTIL the mentioned PR is deployed.

## Risks

I'm not sure this is the right move.  We are potentially introducing a failure mode where on-prem customers now have to enable a certain flag to run correctly, where they didn't before.  However, I don't see a better way to handle this without obscuring errors.

## References

Related to fossas/team-analysis#708

## Checklist

- [x] I added tests for this PR's change (or confirmed tests are not viable).
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [ ] I updated `Changelog.md` if this change is externally facing. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [x] I linked this PR to any referenced GitHub issues, if they exist.

### CODE REVIEW

**Comment by cnr** [comment]
Rather than introducing a new flag, falling back to the old behavior would be nice here. It'll be painful to remove that flag later

**Comment by scruffystuffs** [comment]
I figured we wouldn't remove it, just deprecate it and make it a no-op

**Comment by kitified** [comment]
> Rather than introducing a new flag, falling back to the old behavior would be nice here. It'll be painful to remove that flag later

@thedeerchild, @enricozb - I think we can ensure our monorepo users are on the latest FOSSA version [as soon as it's ready](https://github.com/fossas/FOSSA/pull/6311) so that we don't have to worry about encountering this fallback. Do you agree, or do you think we need this flag?

**Comment by thedeerchild** [comment]
Disagree with the note regarding upgrade cadence, but I think there are few enough that we can manually upgrade them past it. Let's just confirm the Core version somewhere so that we can make sure to upgrade Moto this week (and tell anyone else starting with Monorepo that they need to be above that version).

**Comment by scruffystuffs** [comment]
@thedeerchild Confirming: We're removing the `--monorepo` flag and don't cover the case of out-of-date FOSSA instances here?  If you think that we don't need to cover that case this way I'd prefer to not add the flag.

**Comment by thedeerchild** [comment]
Correct. Cc/ @enricozb

**Comment by scruffystuffs** [comment]
Required FOSSA version is `3.34.19`

**Review by kitified — APPROVED** [review]

**Review by cnr — APPROVED** [review]
LGTM; left a note suggesting to remove the sections added to the userguide

File: docs/userguide.md [hunk 1]
```diff
@@ -437,6 +437,13 @@ By default, `fossa test` displays issues in a human-readable format. To instead
 fossa test --json
 ```
 
+### Testing Monorepo projects
+
+`fossa test` works by fetching the latest "build" from the FOSSA API.  By default, we use that "build" to determine if the project being scanned is a
+standard project, or a monorepo project.  The two project types work differently internally, so we need to know which type we're testing.
+
+In rare cases (like having an out-of-date FOSSA server instance), that info may not be available, and in those cases we assume that the project is a standard project.
+
```
[the remark this answers is not in the document]
Comment (scruffystuffs) [reply docs/userguide.md:446]: Yeah, they made more sense when there was a `--monorepo` flag.

**Pull request merged**